static/box.js

```diff
@@ -1,67 +1,82 @@
-class Shape {
-    /**
-     * Basic shape constructor.
-     * @param x The starting x coordinate.
-     * @param y The starting y coordinate.
-     * @param w The width of the shape.
-     * @param h The height of the shape;
-     * @param fill The color of the shape;
-     */
-    constructor(x, y, w, h, fill) {
-        this.x = x;
-        this.y = y;
-        this.w = w;
-        this.h = h;
-        this.fill = fill;
+
+// This class represents the box at a particular frame of one thing
+class Box {
+
+    constructor(thing, frame, x, y, w, h, interpolated = true) {
+        this.thing = thing; // The parent thing object that owns this keyframe
+        this.frame = frame;
+        this._x = x;
+        this._y = y;
+        this._w = w;
+        this._h = h;
+
+        // true if interpolated, meaning if modified we must insert this back into parent thing as a new keyframe
+        this.interpolated = interpolated;
```

Comment:
Does that mean we're going with every frame has a Box rather than interpolating the Box on the spot?